### PR TITLE
Austin/network

### PR DESCRIPTION
**Pull request by AAAstorga**
Wanting to merge what I have now for networking so Monica can integrate her views. Look in Network.swift for examples. Since we are using static data, code is simple for now.

### CODE REVIEW

File: TCAT/Model/Route.swift [hunk 1]
```diff
@@ -62,12 +65,26 @@ class Route: NSObject {
...
-    var lastStopTime: Date // the critical last time a bus route runs
+    var mainStopsNums: [Int] = [Int]()//-1 for pins
+    var travelDistance: Double = 0.0 //of first stop
+    var lastStopTime: Date = Date()// the critical last time a bus route runs
+    
+   
+    
+    required init(json: JSON) throws {
+        super.init()
+        print(json)
```
Comment (mattbarker016) [inline TCAT/Model/Route.swift]: Maybe comment these out when merging?

File: TCAT/Model/Route.swift [hunk 2]
```diff
@@ -62,12 +65,26 @@ class Route: NSObject {
...
+        departureTime = timeToDate(time: json["departureTime"].stringValue)
+        arrivalTime = timeToDate(time: json["arrivalTime"].stringValue)
+        directions = directionJSON(json:json["directions"].array!)
+        mainStops = json["stopNames"].arrayObject as! [String]
+        mainStopsNums = json["stopNumbers"].arrayObject as! [Int]
+        travelDistance = (directions[0] as! WalkDirection).travelDistance
```
Comment (mattbarker016) [inline TCAT/Model/Route.swift]: Is this a safe assumption? What if there's other walking directions

File: TCAT/Model/Route.swift [hunk 3]
```diff
@@ -78,4 +95,52 @@ class Route: NSObject {
...
+    func directionJSON(json: [JSON]) -> [Direction] {
+        var directionArray = [Direction]()
+        for direction in json {
+            let directionType = direction["directionType"].stringValue
+            if directionType == "walk" {
+                directionArray.append(walkDirection(json: direction))
+            } else if directionType == "depart" {
+                directionArray.append(departDirection(json: direction))
+            } else {
+                directionArray.append(arriveDirection(json: direction))
```
Comment (mattbarker016) [inline TCAT/Model/Route.swift]: I would do 3 else if's and an else to handle errors

File: TCAT/Model/Route.swift [hunk 4]
```diff
@@ -78,4 +95,52 @@ class Route: NSObject {
...
+        let time = timeToDate(time: json["time"].stringValue)
+        let place = json["place"].stringValue
+        let location = CLLocation(latitude: json["location"][0].doubleValue, longitude: json["location"][1].doubleValue)
+        return ArriveDirection(time: time, place: place, location: location)
+        
+    }
+    
+    func timeToDate(time: String) -> Date {
+        let dateFormatter = DateFormatter()
+        dateFormatter.dateFormat = "hh:mm a"
```
Comment (mattbarker016) [inline TCAT/Model/Route.swift]: there is a function in Time.swift that does this string(from time). Switch to it so we can stay consistent.

**Review by mattbarker016 — COMMENTED** [review]
Few stylistic things, plus travelDistance clarification, but switch to the Time.swift function for sure

File: TCAT/Model/Route.swift (hunk 2 above)
Comment (AAAstorga) [reply TCAT/Model/Route.swift]: Actually no it isn't a safe assumption. What does travelDistance mean? The spec in the Route.swift file was a little confusing

Comment (mattbarker016) [reply TCAT/Model/Route.swift]: I think it was supposed to be the "25 minutes" for the entire route in the summary view, but I actually just calculate it from arrival and departure time. We can probably get rid of the variable as a whole

Edit: Oh wait it's travel distance. I think Monica uses it for the 0.1 mi away on the first bus stop. Can you check with her that she uses that variable for that reason?

**Comment by AAAstorga** [comment]
012a36b

**Pull request closed without merging**